Tests/MapboxCoreNavigationTests/MapboxCoreNavigationTests.swift

```diff
@@ -92,43 +116,93 @@ class MapboxCoreNavigationTests: XCTestCase {
         }
     }
     
-    func testNewStep() {
-        // Coordinates from beginning of step[1] to end of step[2]
-        let coordinates = route.legs[0].steps[1].shape!.coordinates + route.legs[0].steps[2].shape!.coordinates
-        let locations: [CLLocation]
-        let now = Date()
-        locations = coordinates.enumerated().map { CLLocation(coordinate: $0.element,
-                                                              altitude: -1, horizontalAccuracy: -1, verticalAccuracy: -1, course: -1, speed: 10,
-                                                              timestamp: now + $0.offset) }
+    func disabled_testNewStep() {
```

Comment:
I've decided to disable `testNewStep` for now. With current implementation of Navigator its original implementation contradicts with other tests (e.g. `testOrderOfExecution` will start failing in case if we use `delay` between each location update and remove this [part](https://github.com/mapbox/mapbox-navigation-ios/blob/release-v2.0/Sources/MapboxCoreNavigation/NavigationService.swift#L324-L330)). Location updates without any delays at this point cause problems as well (Navigator skips certain timestamps even if they're always increasing, `testNotReroutingForAllSteps` is failing because of this).

There is a high chance that `testNotReroutingForAllSteps` will be disabled as well. Even though locally in most of the cases test passes in CircleCI it mostly fails.

Most likely failures in these tests unveil broader problems we currently have in `MapboxCoreNavigationTests`: e.g. Navigator is shared `MapboxNavigationService` between tests, in my opinion tests should be isolated from each other and work independently.

Reply:
Yeah, I've seen other "shared state" kind of issues in other places as well.

Reply:
I also faced the shared state problem when I was implementing NavNative async API, but I manage to solve it by recreating Navigator before every test. So currently there should be no issues with this shared state anymore.
There is still an issue that NavNative skips some callbacks if we send all the locations in one batch, this can be fixed by using ReplayLocationManager with big `speedMultiplier`.